tasks/managed/filter-already-released-advisory-images/filter-already-released-advisory-images.yaml

```diff
@@ -0,0 +1,284 @@
+      type: string
+      default: ""
+  workspaces:
+    - name: data
+      description: Workspace to mount internal request outputs
+  results:
+    - name: result
+      description: Success or failure result
+    - name: filtered_snapshot
+      description: Snapshot JSON with only unpublished images
+    - description: Produced trusted data artifact
+      name: sourceDataArtifact
+      type: string
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /var/workdir
+        name: workdir
+    env:
+      - name: IMAGE_EXPIRES_AFTER
+        value: $(params.ociArtifactExpiresAfter)
+      - name: "ORAS_OPTIONS"
+        value: "$(params.orasOptions)"
+      - name: "DEBUG"
+        value: "$(params.trustedArtifactsDebug)"
+  steps:
+    - name: skip-trusted-artifact-operations
+      ref:
+        resolver: "git"
+        params:
+          - name: url
+            value: $(params.taskGitUrl)
+          - name: revision
+            value: $(params.taskGitRevision)
+          - name: pathInRepo
+            value: stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations.yaml
+      params:
+        - name: ociStorage
+          value: $(params.ociStorage)
+        - name: workDir
+          value: $(params.dataDir)
+    - name: use-trusted-artifact
+      ref:
+        resolver: "git"
+        params:
+          - name: url
+            value: $(params.taskGitUrl)
+          - name: revision
+            value: $(params.taskGitRevision)
+          - name: pathInRepo
+            value: stepactions/use-trusted-artifact/use-trusted-artifact.yaml
+      params:
+        - name: workDir
+          value: $(params.dataDir)
+        - name: sourceDataArtifact
+          value: $(params.sourceDataArtifact)
+    - name: run-script
+      image: quay.io/konflux-ci/release-service-utils:0f82be4be43294b6a96846d87ef7f7c0b9e34267
+      script: |
+        #!/bin/bash
+        set -eux
+
+        SNAPSHOT_FILE="$(params.dataDir)/$(params.snapshotPath)"
+        if [ ! -f "$SNAPSHOT_FILE" ]; then
+            echo "No valid snapshot file was provided at $SNAPSHOT_FILE"
+            exit 1
+        fi
+        snapshot=$(jq -c '.' "$SNAPSHOT_FILE")
+        if [ -z "$snapshot" ]; then
+            echo "Failed to read snapshot file"
+            exit 1
+        fi
+
+        RPA_FILE="$(params.dataDir)/$(params.releasePlanAdmissionPath)"
+        if [ ! -f "$RPA_FILE" ]; then
+            echo "No valid ReleasePlanAdmission file was provided at $RPA_FILE"
+            exit 1
+        fi
+        origin=$(jq -r '.spec.origin' "$RPA_FILE")
+        if [ -z "$origin" ]; then
+            echo "Failed to read origin from ReleasePlanAdmission"
+            exit 1
+        fi
+
+        pipelinerun_label="internal-services.appstudio.openshift.io/pipelinerun-uid"
+
+        # Determine advisory_secret_name based on repositories in snapshot
```

Comment:
May be good to add the stuff from https://github.com/konflux-ci/release-service-catalog/pull/933 in here